EtherealEngine/etherealengine
# Mocap pose based interactions

## Summary
Lays the groundwork for an extensible pose based interaction system. The first mocap pose based event included in this PR is seat detection via interaction actions recieved by mount point components. Sitting with full body motion capture next to a seat will run the mount point component's seated logic. Standing up should also cause the avatar to leave the seat.

The interaction action/onInteract function has been replaced in favor of directly accessing input sources through the input source component, this also required some code changes in the grabbable system. In a similar vein, the local pose component is used to manage mocap pose inputs after pose detection. For now, the only integrated poses are sitting and standing, though pose types can be extended to any theoretically detectable pose. Pose detection for the seated/standing position is done through a reusable function that compares upper leg angles relative to the hips.

## References
closes #_insert number here_

## QA Steps

- Go to studio/default-project/apartment
- Add an entity with a mount point component
- Visit location/apartment
- Stand near the mount point component in the location
- Open the capture page
- Sit down in view of the camera
- Avatar should snap to the mount
- Stand up
- Avatar should dismount
